### PR TITLE
Enforced authentication

### PR DESCRIPTION
**Pull request by a-nersharma**
# stream-registry PR

Enforced Authentication

### Changed
Enhanced Authentication in cli

# PR Checklist Forms

- [x] CHANGELOG.md updated
- [x] Reviewer assigned
- [ ] PR assigned (presumably to submitter)
- [ ] Labels added (enhancement, bug, documentation)

### CODE REVIEW

File: state/graphql-sender/src/main/java/com/expediagroup/streamplatform/streamregistry/state/graphql/DefaultApolloClientFactory.java [hunk 1]
```diff
@@ -34,12 +33,6 @@
   private Credentials credentials;
   @NonNull private final Consumer<OkHttpClient.Builder> configurer;
 
-  public DefaultApolloClientFactory(String streamRegistryUrl) {
```
Comment (nahguam) [inline state/graphql-sender/src/main/java/com/expediagroup/streamplatform/streamregistry/state/graphql/DefaultApolloClientFactory.java:37]: For OSS users, they may still wish to use anonymous so these should remain.

File: cli/src/main/java/com/expediagroup/streamplatform/streamregistry/cli/command/delete/EntityClient.java [hunk 2]
```diff
@@ -191,9 +192,11 @@
 
   static class Factory {
     @Option(names = "--streamRegistryUrl", required = true) String streamRegistryUrl;
+    @Option(names = "--username", required = true) String username;
```
Comment (nahguam) [inline cli/src/main/java/com/expediagroup/streamplatform/streamregistry/cli/command/delete/EntityClient.java:195]: For OSS users, both here and in `GraphQLEventSenderAction` `username` and `password` should be optional.

They'd also be better named as `streamRegistryUsername` and `streamRegistryPassword` as they're used as a `@Mixin` with other options in the CLI so it should be clear what they are the username and password for.

File: cli/src/main/java/com/expediagroup/streamplatform/streamregistry/cli/action/GraphQLEventSenderAction.java [hunk 3]
```diff
@@ -22,16 +22,23 @@
...
+  @Option(names = "--streamRegistryUsername", required = false)
+  @Getter String streamRegistryUsername;
+
+  @Option(names = "--streamRegistryPassword", required = false)
+  @Getter String streamRegistryPassword;
+
   @Override
   public EventSender sender() {
-    ApolloClient client = new DefaultApolloClientFactory(streamRegistryUrl).create();
+    ApolloClient client = new DefaultApolloClientFactory(streamRegistryUrl, new Credentials(streamRegistryUsername, streamRegistryPassword)).create();
```
Comment (nahguam) [inline cli/src/main/java/com/expediagroup/streamplatform/streamregistry/cli/action/GraphQLEventSenderAction.java:41]: If they're now optional, you should only provide a credentials object if they've been provided.

File: state/graphql-sender/src/main/java/com/expediagroup/streamplatform/streamregistry/state/graphql/DefaultApolloClientFactory.java (hunk 1 above)
Comment (a-nersharma) [reply state/graphql-sender/src/main/java/com/expediagroup/streamplatform/streamregistry/state/graphql/DefaultApolloClientFactory.java]: done

File: cli/src/main/java/com/expediagroup/streamplatform/streamregistry/cli/action/GraphQLEventSenderAction.java (hunk 3 above)
Comment (a-nersharma) [reply cli/src/main/java/com/expediagroup/streamplatform/streamregistry/cli/action/GraphQLEventSenderAction.java]: done

**Review by nahguam — APPROVED** [review]

**Pull request merged**